nilearn/datasets/_utils.py

```diff
@@ -347,31 +347,31 @@ def uncompress_file(file_, delete_archive=True, verbose=1):
             z.extractall(path=data_dir)
             z.close()
             if delete_archive:
-                os.remove(file_)
-            file_ = filename
+                file_.unlink()
+            file_ = Path(filename)
             processed = True
         elif file_.suffix == ".gz" or header.startswith(b"\x1f\x8b"):
             import gzip
 
             if file_.suffix == ".tgz":
-                filename = Path(f"{filename}.tar")
+                filename = f"{filename}.tar"
```

Comment:
```suggestion
                filename = filename.with_suffix(".tar")
```
double check but I think this should work